botorch/optim/initializers.py

```diff
@@ -169,10 +236,18 @@ def gen_batch_initial_conditions(
             else:
                 X_rnd = (
                     get_polytope_samples(
-                        n=n * q,
-                        bounds=bounds,
-                        inequality_constraints=inequality_constraints,
-                        equality_constraints=equality_constraints,
+                        n=n,
+                        bounds=torch.hstack([bounds for _ in range(q)]),
```

Comment:
So the whole point of this change is to draw a `n` samples from a `q*d`-dim space, rather than `n*q` samples from a `d`-dim space. I don't have a great sense of this, but one thing that may happen by doing os is that you may need to increase the burnin and thinning options here to get comparable mixing behavior for the chain. I haven't really looked into this much, but it would be worth trying to do a basic numerical study that compares this (without actually imposing the cross-batch constraints) just to see how much of an issue this is.

Reply:
Good point, I did some easy test and looked at the autocorrelation of the samples and indeed it has an influence:

In the old way, I get for a scenario with a q-batch of 4 and 1024 raw samples the following autocorrealtion:

``` python
import numpy as np
import torch
from botorch.utils.sampling import get_polytope_samples
from botorch.optim.initializers import transform_constraints
import matplotlib.pyplot as plt


tkwargs = {'dtype': torch.float64, 'device': 'cpu'}

bounds = torch.tensor([[0,0,0,0,0], [1,1,1,1,1]]).to(**tkwargs)

equalities = [
    (torch.tensor([0,1,2,3], dtype=torch.int64), torch.tensor([1,1,1,1]).to(**tkwargs), 1.)
]

inequalities = [
    (torch.tensor([0,1], dtype=torch.int64), torch.tensor([1,1]).to(**tkwargs), 0.3)
]

q = 4
raw_samples=1024

samples = get_polytope_samples(
    n=q*raw_samples,
    bounds=bounds,
    inequality_constraints=inequalities,
    equality_constraints=equalities,
    n_burnin=10000,
    thinning=32
)

samples = samples.detach().numpy()

plt.acorr(samples[:,0], maxlags=24,normed=True)
plt.show()
```

![image](https://user-images.githubusercontent.com/28678090/227361944-9d982165-b23d-4a43-980a-710cb93322b4.png)

Doing it the new way via sampling from a q*d space, I get the following:

``` python
from botorch.optim.initializers import transform_constraints

samples2 = get_polytope_samples(
    n=raw_samples,
    bounds=torch.hstack([bounds for _ in range(q)]),
    inequality_constraints=transform_constraints(inequalities,q=q,d=5),
    equality_constraints=transform_constraints(equalities,q=q,d=5),
    n_burnin=10000,
    thinning=32
)

samples2 = samples2.reshape(torch.Size((q*raw_samples, 5)))

samples2 = samples2.detach().numpy()

plt.acorr(samples2[:,0], maxlags=24,normed=True)
plt.show()
```
![image](https://user-images.githubusercontent.com/28678090/227362358-2d68c146-bd73-4a30-af1c-94ecfd515a57.png)

We see higher correlation for every fourth sample, which makes sense as four samples here correspond to one polytope sample in the new setup. Setting `thinning` to `32*q` cures the problem:

![image](https://user-images.githubusercontent.com/28678090/227363058-4b670a7e-c034-430e-ab7c-67ff63dfa2c7.png)

So I changed the default to always 32*q.

What do you think? Is this sufficient?

Reply:
Actually, one concern here is that you are transforming these constraints regardless of whether there are any inter-point / cross-batch constraints present. That means that in that case we're running the hit-and-run sampler in an unnecessarily high-dimensional space, and could instead just run it for `n*q` samples as we did before. Could you please modify this logic here to do this branching? Would be good to move this whole branch starting in l 257 out into a helper function to keep things tidy.